You are given a number n. You need to generate and print a pattern based on the given value of n.

For each row, starting from the first, print numbers in descending order from n down to 1.
Each number in a row is repeated as many times as the current row index (starting from n).
Instead of printing each row on a new line, separate rows with -1.
Instead of a newline at the end of each row, print -1 to indicate row separation. After printing the entire pattern, end the output with -1.

For n= 3,
pattern:  3 3 3 2 2 2 1 1 1
                    3 3 2 2 1 1 
                    3 2 1

For n= 2,
pattern:  2 2 1 1
                    2 1

Examples :

Input: 2
Output: [2, 2, 1, 1, -1, 2, 1, -1]
Input: 3
Output: [3, 3, 3, 2, 2, 2, 1, 1, 1, -1, 3, 3, 2, 2, 1, 1, -1, 3, 2, 1, -1]
Input: 1
Output: [1, -1]

In [11]:
n=int(input("Enter the number : "))
out=[]
for i in range(n,0,-1):
   for j in range(n,0,-1):
        x=[j]*i
        out+=x
   out.append(-1)
print((out))



[3, 3, 3, 2, 2, 2, 1, 1, 1, -1, 3, 3, 2, 2, 1, 1, -1, 3, 2, 1, -1]


Given a number n, print a sequence of numbers starting from n. Each next number in the sequence is n - 5, and this continues recursively until the number becomes less than or equal to 0. After that, print the sequence in reverse order, adding 5 each time, until it reaches back to the original number n.
Note: You must not use loops.

Examples:

Input: n = -16
Output: [-16]
Explanation: Since -16 is less than zero so it will remain same.
Input: n = 10
Output: [10, 5, 0, 5, 10]
Explanation: The value decreases until it is greater or equal to 0. After that it increases and stops when it becomes 10 again.

In [ ]:
def neg(n,a):

    if n>0:
        print(n, end=" ")
        neg(n-5,a)
    elif n==0:
        print(n, end=" ")
        pos(n+5,a)
        
def pos(n,a):
    if n==a:
        print(n, end=" ")
    else:
        print(n,end=" ")
        pos(n+5,a)

n=int(input("Enter the number : "))
a=n
neg(n,a)



15 10 5 0 5 10 15 

In this task, we would like for you to appreciate the usefulness of the groupby() function of itertools . To read more about this function, Check this out .

You are given a string . Suppose a character '' occurs consecutively  times in the string. Replace these consecutive occurrences of the character '' with  in the string.

For a better understanding of the problem, check the explanation.

Input Format

A single line of input consisting of the string .

Output Format

A single line of output consisting of the modified string.

Constraints

All the characters of  denote integers between  and .


Sample Input

1222311
Sample Output

(1, 1) (3, 2) (1, 3) (2, 1)

In [ ]:
str=input("Enter your string : ")
l={}
count=0
for i in str:
    l[i]=l.get(i,0)+1

print(l)






{'1': 3, '2': 3, '3': 1}


IMPORTANT QUESTIONS FOR PYTHON 

1. Check Whether a Number is Even or Odd  
  
Problem Statement :  
Given an integer N, determine whether the given number is even or odd. A number is even if it is completely divisible by 2. Otherwise, it is odd.

In [2]:
n=int(input("Enter a number : "))
if n%2==0:
    print(f"The given {n} is even.")
else:
    print(f"The given {n} is  odd.")

The given 24 is even.


2. Check Whether a Number is Positive, Negative, or Zero  
  
Problem Statement:  
Given an integer N, determine whether it is positive, negative, or zero.

In [4]:
n=int(input("Enter any number : "))
if n>0:
    print(f"The given {n} is positive.")
elif n<0:
    print(f"The given {n} is neagtive.")
else:
    print(f"The given {n} is Zero.")

The given -8 is neagtive.


3. Find the Greatest of Two Numbers 
   
Problem Statement:  
Given two integers, find the greater number.

In [6]:
n1=int(input("Enter first number : "))
n2=int(input("Enter second number : "))
if n1>n2:
    print(f"{n1} is greater than {n2}.")
else:
    print(f"{n2} is greater than {n1}")

23 is greater than 12.


4. Find the Greatest of Three Numbers  
  
Problem Statement:  
Given three integers, find the greatest among them.

In [10]:
n1=int(input("Enter first number : "))
n2=int(input("Enter second number : "))
n3=int(input("Enter third number : "))
if (n1>n2) and (n1>n3):
    print(f"{n1} is greater than both {n2} and {n3}.")
elif n2>n3:
    print(f"{n2} is greater than {n1} and {n3}.")
else:
    print(f"{n3} is greater than {n1} and {n2}.")


15 is greater than 12 and 14.


5. Check Whether a Year is a Leap Year  
  
Problem Statement:  
Given a year, determine whether it is a leap year. A year is a leap year if it is divisible by 400, or if it is divisible by 4 but not divisible by 100.

In [13]:
year=int(input("Enter your year : "))
if (year%400==0 or year%4==0) and (year%100!=0):
    print(f"The given {year} is Leap year.")
else:
    print(f"The given {year} is not Leap year.")

The given 1900 is not Leap year.


6. Find the Sum of First N Natural Numbers  
  
Problem Statement:  
Given a positive integer N, find the sum of the first N natural numbers.

In [16]:
n=int(input("Enter any number : "))
sum=0
for i in range(1,n+1):
    sum+=i

print(f"Sum of all numbers : {sum}")

Sum of all numbers : 630


7. Count the Number of Digits in an Integer  
  
Problem Statement:  
Given an integer N, count how many digits it contains.

In [ ]:
"""Using While Loop"""
n=int(input("Enter your number : "))
count=0
while n>0:
    count+=1
    n=n//10
print(f"no. of digit : {count}")

"""Using normal method"""
n=int(input("Enter your number : "))
print(f"NO. of digits : {len(str(abs(n)))}")



NO. of digits : 11


8. Find the Sum of Digits of a Number  
  
Problem Statement:  
Given an integer N, find the sum of all its digits.

In [24]:
n=int(input("Enter the number : "))
sum=0
while n>0:
    rem=n%10
    sum+=rem
    n//=10

print(f"Sum of all digits which are present in number : {sum}")

Sum of all digits which are present in number : 2


9. Reverse the Digits of a Number  
  
Problem Statement:  
Given an integer N, reverse its digits.

In [ ]:
"""Using loop method """
n=int(input("Enter your number : "))
rev=0
while n>0:
    rem=n%10
    rev=rev*10+rem
    n//=10
print(f"Reverse of number : {rev}")

"""Using Slicing method. """
n=int(input("Enter your number : "))
s=str(n)
print(f"Reverse of number : {s[::-1]}")


Reverse of number : 00401


10. Check Whether a Number is a Palindrome  
  
Problem Statement:  
Given an integer N, determine whether it reads the same forward and backward.

In [35]:
"""USING SCICING METHOD """
# n=int(input("Enter your number : "))
# a=n
# s=str(n)
# rev=int(s[::-1])

# if a==rev:
#     print(f"{a} is Palindrome number.")
# else:
#     print(f"{a} is not Palindrome number.")

"""USING LOOP METHOD """
n=int(input("Enter any number : "))
a=n
rev=0
while n>0:
    rem=n%10
    rev=rev*10+rem
    n//=10

if a==rev:
    print(f"{a} is Palindrome number.")
else:
    print(f"{a} is not Palindrome number.")



12345 is not Palindrome number.


11. Find the Maximum and Minimum Digit in a Number  
  
Problem Statement:  
Given an integer N, find the largest and smallest digit present in the number.

In [36]:
n=int(input("Enter any number : "))
l=[]
while n>0:
    rem=n%10
    l.append(rem)
    n//=10

print(f"The maximum digit {max(l)} and minimum digit {min(l)} in number.")


The maximum digit 9 and minimum digit 0 in number.


12. Count the Occurrences of a Given Digit  
  
Problem Statement:  
Given an integer N and a digit D, count how many times digit D occurs in N.

In [42]:
n=int(input("Enter any number : "))
d=input("Digit :")

out=""
while n>0:
    rem=n%10
    out+=str(rem)
    n//=10

if d in out:
    print(f"Occurance of digit is {out.count(str(d))}.")
else:
    print(f"The digit is not present in number.")


The digit is not present in number.


13. Find the Frequency of Every Digit  
  
Problem Statement:  
Given an integer N, find how many times each digit from 0 to 9 occurs.

In [7]:
n=int(input("Enter your number : "))
d="0123456789"
out=""
while n>0:
    rem=n%10
    out=out+str(rem)
    n//=10

for i in d:
    if i in out:
        print(f"{i}--> {out.count(i)},",end=" ")
    elif i=="0":
        print(f"{i}--> {out.count(i)},",end=" ")
    else:
        print(f"and all digits--> 0, ",end=" ")
        break
      


0--> 0, 1--> 1, 2--> 2, 3--> 1, 4--> 2, 5--> 2, 6--> 1, 7--> 2, 8--> 1, 9--> 1, 

14. Replace All 0s with 1s in a Number  
  
Problem Statement:  
Given an integer N, replace every 0 digit with 1.

In [10]:
n=int(input("Enter any number : "))
s=str(n)
print(int(s.replace("0","1")))

214


15. Remove a Given Digit from a Number  
  
Problem Statement:  
Given an integer N and a digit D, remove all occurrences of D from the number.

In [16]:
"""Method-1"""
# n=int(input("Enter any number : "))
# d=input("Enter any digit which you want to delete : ")

# s=str(n)
# for i in s:
#     if i==d:
#         continue 
#     else:
#         print(int(i),end="")

"""Method-2"""
n=int(input("Enter any number : "))
d=int(input("Enter any digit which you want to delete : "))
l=[]
while n>0:
    rem=n%10
    l.append(rem)
    n//=10

s=l.remove(d)
# v="".join(s)
print(s)





None


16. FIND THE PRODUCT OF DIGITS OF A NUMBER. 
  
PROBLEM STATEMENT:  
GIVEN AN INTEGER N, FIND THE  PRODUCT OF ALL ITS DIGITS.


In [1]:
n=int(input("Enter any number : "))
mul=1
while n>0:
    rem=n%10
    mul*=rem
    n//=10

print(f"Multiply of all digits : {mul}")

Multiply of all digits : 24


17. REPEATEDLY ADD DIGITS UNTIL A SINGLE DIGIT REMAINS.  
  
PROBELM SYSTEM:  
GIVEN AN INTEGER N, REPEATEDLY ADD ALL ITS DIGITS UNTIL THE RESULT CONTAINS ONLY  ONE DIGIT.

In [1]:
n=int(input("Enter any number : "))
sum=0
while n>0:
    rem=n%10
    sum+=rem
    n//=10
n=sum
sum=0
while n>0:
    rem=n%10
    sum+=rem
    n//=10

print(sum)


7


18. Find the Sum of Squares of Digits  
  
Problem Statement:  
Given an integer N, calculate the sum of the squares of all its digits.

In [2]:
n=int(input("Enter any number : "))
sum=0
while n>0:
    rem=n%10
    sum+=rem**2
    n//=10

print(f"Sum of square of digits : {sum}")

Sum of square of digits : 14


19. Check Whether a Number is Prime  
  
Problem Statement:  
Given an integer N, determine whether it is a prime number. A prime number is greater than 1 and has exactly two factors: 1 and itself.

In [7]:
n=int(input("Enter any number : "))
count=0
for i in range(2,n+1):
    if n%i==0:
        count+=1

if count==1:
    print(f"The given {n} is prime.")
else:
    print(f"The given {n} is not prime.")

The given 37 is prime.


20. Print All Prime Numbers in a Given Range.  
  
Problem Statement:  
Given two integers L and R, print all prime numbers between them.

In [6]:
n1=int(input("Enter the starting point : "))
n2=int(input("Enter the ending point : "))

for i in range(n1,n2+1):
    count=0
    for j in range(1,i+1):
        if i%j==0:
            count+=1

    if count==2:
        print(f"{i}",end=" ")   


2 3 5 7 11 13 17 19 23 29 31 37 41 43 47 53 59 61 67 71 73 

21. Find the Prime Factors of a Number  
  
Problem Statement:  
Given an integer N, print all its prime factors.

In [ ]:
n=int(input("Enter any number : "))
l=[]
d=2

while n>1:
    while n%d==0:
        l.append(d)
        n//=d
    d+=1
print(l)

[2, 2, 3, 5]


22. Express a Number as the Sum of Two Prime Numbers  
  
Problem Statement:  
Given an integer N, determine whether it can be expressed as the sum of two prime numbers

In [23]:
n=int(input("Enter any number : "))
l=[]
for i in range(2,n):
    count=0
    for j in range(1,i+1):
        if i%j==0:
            count+=1
    if count==2:
        l.append(i)

# print(l)

f=0
for i in range(0,len(l)):
    for j in range(0,len(l)):
        if (l[i]+l[j]==n):
            print(l[i],l[j])
            f=1

if f==1:
    print("True")
else:
    print("False")

False


23. Check Whether a Number is an Armstrong Number  
  
Problem Statement:  
Given an integer N, determine whether it is an Armstrong number. For an N-digit number, calculate the sum of each digit raised to the power of the number of digits. If the sum equals the original number, it is an Armstrong number.

In [9]:
n=int(input("Enter any number : "))
a=n
sum=0

while n>0:
    rem=n%10
    sum=sum+rem**3
    n//=10

if sum==a:
    print(f"The given {a} is Armstrong number.")
else:
    print(f"The given {a} is not Armstrong number.")

The given 132 is not Armstrong number.


24. Check Whether a Number is a Perfect Number  
  
Problem Statement:  
A number is called a Perfect Number if the sum of its proper divisors is equal to the number itself

In [16]:
n=int(input("Enter any number : "))
sum=0
for i in range(1,n):
    if n%i==0:
        sum+=i

if sum==n:
    print(f"The given {n} is perfect number.")
else:
    print(f"The given {n} is not perfect number.")


The given 496 is perfect number.


25. Check Whether a Number is an Abundant Number  
  
Problem Statement:  
A number is called an Abundant Number if the sum of its proper divisors is greater than the number itself.

In [ ]:
n=int(input("Enter any number : "))
sum=0
for i in range(1,n):
    if n%i==0:
        sum+=i

if sum>n:
    print(f"Given number {n} and sum of all divisor {sum}, So the given number is Abundant number.")
else:
    print(f"Given number {n} and sum of all divisor {sum}, So the given number is not Abundant number.")


Given number 28 and sum of all divisor 28, So the given number is not Abundant number.


26. Check Whether a Number is Perfect, Abundant, or Deficient  
  
Problem Statement:  
Given an integer N, classify it based on the sum of its proper divisors. Sum = N → Perfect, Sum > N → Abundant, Sum < N → Deficient.

In [25]:
n=int(input("Enter any number : "))
a=n//2
sum=0
for i in range(1,a+1):
    if n%i==0:
        sum+=i

if sum==n:
    print(f"The given {n} is Perfect number.")
elif sum>n:
    print(f"The given {n} is Abundant number.")
else:
    print(f"The given {n} is Deficient number.")

The given 10 is Deficient number.


27. Check Whether a Number is a Strong Number  
  
Problem Statement:  
A Strong Number is a number in which the sum of the factorials of its digits is equal to the original number

In [32]:
n=int(input("Enter any number : "))
a=n
sum=0
mul=1
while n>0:
    rem=n%10
    for i in range(1,rem+1):
        mul*=i
    sum+=mul
    mul=1
    n//=10

if a==sum:
    print(f"The given {a} is Strong number.")
else:
    print(f"The given {a} is not Strong number.")


The given 145 is Strong number.


28. Check Whether a Number is a Harshad Number  
  
Problem Statement:  
A number is called a Harshad Number (or Niven Number) if it is divisible by the sum of its digits.

In [36]:
n=int(input("Enter any number : "))
a=n
sum=0
while n>0:
    rem=n%10
    sum+=rem
    n//=10

if a%sum==0:
    print(f"The given {a} is harshad number.")
else:
    print(f"The given {a} is not harshad number.")

The given 378 is harshad number.


29. Check Whether a Number is an Automorphic Number  
  
Problem Statement:  
A number is called Automorphic if its square ends with the number itself.

In [47]:
n=int(input("Enter any number : "))
a=n%10
sq=a**2
if (a==sq%10):
    print(f"{n} is Automorphic number.")
else: 
    print(f"{n} is not Automorphic number.")

76 is Automorphic number.


30. Find the Factorial of a Number  
  
Problem Statement:  
Given a non-negative integer N, find its factorial. Factorial is defined as N! = N × (N-1) × ... × 2 × 1

In [ ]:
n=int(input("Enter any number : "))
mult=1


for i in range(1,n+1):
    mult*=i

print(f"factorial of a number : {mult}")

factorial of a number : 720


31. Calculate the Power of a Number  
  
Problem Statement:  
Given two integers X and N, calculate X raised to the power N

In [ ]:
n=int(input("Enter any number: "))
p=int(input("How many power do you want ? "))

print(f"Number : {n**p}")

Number : 1024


32. Find the Sum of an Arithmetic Progression  
  
Problem Statement:  
Given the first term a, common difference d, and number of terms n, find the sum of the AP

In [ ]:
a=int(input("Enter first number : "))
d=int(input("Enter the common difference : "))
n=int(input("NO. of Terms : "))

s=(2*a+(n-1)*d)*(n//2)
print(f"Sum of an Arthmetic Progression: {s}")

"""Another methods """

a=int(input("Enter first number : "))
d=int(input("Enter the common difference : "))
n=int(input("NO. of Terms : "))
sum=0
i=1
while i<=n:
    print(a,end=" ")
    sum+=a
    a+=d
    i+=1

print(f"\n Sum of all number : {sum}")

Sum of an Arthmetic Progression: 156


33. Find the Sum of a Geometric Progression  
  
Problem Statement:  
Given the first term a, common ratio r, and number of terms n, find the sum of the GP

In [ ]:
a=int(input("Enter the first term : "))
r=int(input("Enter the common ration : "))
n=int(input("Enter no. of terms : "))

sum=0
i=1
while(i<=n):
    print(a,end=" ")
    sum+=a
    a=a*r
    i+=1

print(f"\nSum of a Geometric progression : {sum}")


2 6 18 54 Sum of a Geometric progression : 80


34. Print the Fibonacci Series  
  
Problem Statement:  
Print the first N Fibonacci numbers. The sequence starts with 0, 1, 1, 2, 3, 5, 8, ... Each term is the sum of the
previous two terms.


In [19]:
n=int(input("Enter the no. of terms : "))
a=0
b=1
i=1
print(a,b,end=" ")
while(i<=n-2):
    c=a+b
    print(c,end=" ")
    a,b=b,c
    i+=1



0 1 1 2 3 5 8 13 

35. Find the GCD of Two Numbers  
  
Problem Statement:  
Given two integers A and B, find their Greatest Common Divisor (GCD).

In [22]:
n1=int(input("Enter first number : "))
n2=int(input("Enter second number : "))
small=0
if n1>n2:
    small=n2
else:
    small=n1

for i in range(1,small+1):
    if n1%i==0 and n2%i==0:
        gcd=i

print(f"The greatest common divisor of {n1} and {n2} is {gcd}.")

The greatest common divisor of 1255 and 335 is 5.


36. Find the LCM of Two Numbers  
  
Problem Statement:  
Given two integers A and B, find their Least Common Multiple (LCM).

In [24]:
n1=int(input("Enter the first number : "))
n2=int(input("Enter the second number : "))
small=0
if n1>n2:
    small=n2
else:
    small=n1

for i in range(1,small+1):
    if n1%i==0 and  n2%i==0:
        hcf=i

lcm=(n1*n2)//hcf
print(f"{lcm} is the smallest positive number divisble by both {n1} and {n2}.")

75 is the smallest positive number divisble by both 15 and 25.


37. Count the Number of Factors of a Number  
  
Problem Statement:  
Given an integer N, find the total number of positive factors of N.

In [27]:
n=int(input("Enter any number : "))
count=0
for i in range(1,n+1):
    if n%i==0:
        count+=1
        print(i, end=" ")
print(f"\nNumber of factors : {count}")


1 2 29 58 
Number of factors : 4


38. Find the Sum of All Factors of a Number  
  
Problem Statement:  
Given an integer N, find the sum of all its positive factors

In [ ]:
n=int(input("Enter any number : "))
sum=0
for i in range(1,n+1):
    if n%i==0:
        sum+=i
        print(i, end=" ")
print(f"\nSum of all factors : {sum}")
    

1 2 3 4 6 12 Sum of all factors : 28


39. Check Whether Two Numbers are Co-Prime  
  
Problem Statement:  
Two numbers are called co-prime if their GCD is 1. Given two integers, determine whether they are co-prime.


In [31]:
n1=int(input("Enter the first number : "))
n2=int(input("Enter the  second number : "))
small=0
if n1>n2:
    small=n2
else:
    small=n1

count=0
for i in range(1,small+1):
    if n1%i==0 and n2%i==0:
        count+=1

if count==1:
    print(f"Both the given numbers are co-prime.")
else:
    print(f"Both the given numbers are not co-prime.")


Both the given numbers are not co-prime.


40. Find the GCD of an Array of Numbers  
  
Problem Statement:  
Given an array of integers, find the GCD of all its elements.

In [6]:
pmt="How many elements do you want in array : "
n=int(input(pmt))
l=[]
for i in range(1,n+1):
    l.append(eval(input(f"Element {i} : ")))

print("List : ",l)
a=min(l)
for i in range(1,a):
        if l[0]%i==0:
            gcd=i
print(f"GCD of list : {gcd}")



List :  [12, 18, 24]
GCD of list : 6


41. Check Whether a Number is a Neon Number  
  
Problem Statement:  
A number is called a Neon Number if the sum of the digits of its square is equal to the number itself

In [13]:
n1=int(input("Enter any number : "))
a=n1
n2=n1**2
sum=0
while n2>0:
    rem=n2 % 10
    sum+=rem
    n2//=10

if a==sum:
    print(f"The given {a} number is neon number.")
else:
    print(f"The given {a} number is not neon number.")



The given 1 number is neon number.


42. Check Whether a Number is a Spy Number  
  
Problem Statement:  
A number is called a Spy Number if the sum of its digits is equal to the product of its digits.

In [15]:
n=int(input("Enter any number : "))
sum=0
mul=1
while n>0:
    rem=n%10
    sum+=rem
    mul*=rem
    n//=10
print(f"Sum and Product of all digits are {sum} and {mul} respectively.")

if sum==mul:
    print(f"The given number is Spy number.")
else:
    print(f"The given number is not Spy number.")

Sum and Product of all digits are 13 and 60 respectively.
The given number is not Spy number.


43. Check Whether a Number is a Duck Number.  
  
Problem Statement:  
A Duck Number is a number that contains at least one zero in its digits.


In [23]:
n=int(input("Enter any number : "))
count=0
while n>0:
    rem=n%10
    if rem==0:  
        count=1
    n//=10

if count==1:
    print("The given number is Duck number.")
else: 
    print("The given number is not Duck number.")
    
     



The given number is Duck number.


44. Check Whether a Number is a Disarium Number.  
  
Problem Statement:  
A number is called a Disarium Number if the sum of its digits raised to their respective positions equals the
number. Positions start from 1 from the left.

In [34]:
n=int(input("Enter any number : "))
a=n
s=str(n)
mul=1
sum=0
for i in range(0,len(s)):
    mul=int(s[i])**(i+1)
    sum+=mul

if a==sum:
    print(f"The given {a} number is Disarium number.")
else:
    print(f"The given {a} number is not Disarium number.")


The given 135 number is Disarium number.


45. Check Whether a Number is a Happy Number.  
   
Problem Statement:  
A number is called a Happy Number if repeatedly replacing the number by the sum of the squares of its digits
eventually produces 1. If the process enters a cycle without reaching 1, it is not happy

In [4]:
n=int(input("Enter any number : "))
a=n
s=[]
f=0
while True :
    sum=0
    while n>0:
        rem=n%10
        sum=sum+rem**2
        n//=10
    if sum==1:
        f=1
        break
    elif sum in s:
        break
    else:
        s.append(sum)
        n=sum


if f==1:
    print(f"The given {a} number is Happy number.")
else:
    print(f"The given {a} number is not Happy number.")

The given 15 number is not Happy number.


46. Calculate the Area of a Circle  
  
Problem Statement:  
Given the radius of a circle, calculate its area

In [6]:
r=int(input("Enter the radius : "))
a=3.14*(r**2)
print(f"Area of Circle : {a}")

Area of Circle : 314.0


47. Find the Roots of a Quadratic Equation  
  
Problem Statement:  
Given the coefficients a, b, and c of ax² + bx + c = 0, find its roots. Here a must not be zero

In [12]:
a=int(input("Enter the value of a :"))
b=int(input("Enter the value of b : "))
c=int(input("Enter the value of c : "))

x1=((-b)+(b**2-4*a*c)**0.5)//2*a
x2=((-b)-(b**2-4*a*c)**0.5)//2*a


d=(b**2)-4*a*c
print(f"Roots : {x1},{x2}")
if d>0:
    print(f"Real and different roots.")
elif d==0:
    print(f"One real root.")
else:
    print(f"Complex roots.")



Roots : 5.0,-2.0
Real and different roots.


48. Calculate Permutations: N People Occupying R Seats.  
  
Problem Statement:  
Given N people and R seats, find the number of possible arrangements when order matters.


In [17]:
def factorial(n):
    if n==0 or n==1:
        return 1
    return n*factorial(n-1)


n=int(input("Enter no. of people : "))
r=int(input("Enter the no. of seats : "))

nPr=factorial(n)//factorial(n-r)

print(f"No. of possible  arrangement : {nPr}")

No. of possible  arrangement : 360


49. Calculate Combinations: Select R People from N People.  

Problem Statement:  
Given N people and R selections, find the number of ways to select R people from N people when order does
not matter.

In [23]:
def fact(n):
    if n==0 or n==1:
        return 1
    return n*fact(n-1)


n=int(input("Enter the no. of people : "))
r=int(input("Enter the no. of selection : "))

nCr=fact(n)//(fact(r)*fact(n-r))

print(f"No. of ways : {nCr}")

No. of ways : 15


50. Add Two Fractions and Simplify the Result.  
  
Problem Statement:  
Given the numerator and denominator of two fractions, calculate their sum and return the result in its simplest
form.

In [25]:
n1=int(input("Enter the numerator1 : "))
d1=int(input("Enter the denominator1 : "))
n2=int(input("Enter the numerator2 : "))
d2=int(input("Enter the denominator2 : "))

num=(n1*d2)+(n2*d1)
dem=d1*d2

if num>dem:
    small=dem
else:
    small=num

for i in range(1,small+1):
    if num%i==0 and dem%i==0:
        num=num//i
        dem=dem//i
print(f"Numerator : {num} and Denominator : {dem}")


Numerator : 6 and Denominator : 2


Plus one  
Example 1:  
Input: digits = [1,2,3]  
Output: [1,2,4]  
Explanation: The array represents the integer 123.  
Incrementing by one gives 123 + 1 = 124.  
Thus, the result should be [1,2,4].  

Example 2:  
Input: digits = [9]  
Output: [1,0]  
Explanation: The array represents the integer 9.  
Incrementing by one gives 9 + 1 = 10.  
Thus, the result should be [1,0].  

In [2]:
l=eval(input("Enter your list : "))
s=[]
sum=0
if len(l)==1:
    sum=l[0]+1
    while sum>0:
        rem=sum%10
        s.append(rem)
        sum//=10
    s.reverse()
    print(f"output: {s}")
else:
    l[-1]=l[-1]+1
    print(f"Output: {l}")

output: [2, 0]


In [2]:
a=input("Enter first word : ")
b=input("Enter second word : ")

if len(a)<len(b):
    small=a
    long=b
else:
    small=b
    long=a

print(small+long+small)

aycakuhwbcbayc


Write a function called highest_frequency that takes a list of integers as input and returns a list of integers that have the highest frequency. If multiple integers have the same highest frequency, all of them should be included in the list.

For example, given the list [1, 3, 2, 2, 1, 1, 4, 2, 4, 3], the function should return [1, 2], as both 1 and 2 appear 3 times in the list, which is the highest frequency among all integers.

In [1]:
def highest_frequency(lst):
	frequency = {}
	max_freq = 0

	# Count the frequency of each element in the list
	for num in lst:
		frequency[num] = frequency.get(num, 0) + 1
		max_freq = max(max_freq, frequency[num])

	# Find all numbers with the highest frequency
	result = []
	for num, freq in frequency.items():
		if freq == max_freq:
			result.append(num)

	return result

print(highest_frequency([1,2,3,2,1]))

[1, 2]


Write a function called "multiplication_table" that accepts a size parameter. Return a nested array to represent the table.  
  
Example: multiplication_table(5)  
[[0, 1, 2,  3,  4,  5],  
 [1, 1, 2,  3,  4,  5],  
 [2, 2, 4,  6,  8,  10],  
 [3, 3, 6,  9,  12, 15],  
 [4, 4, 8,  12, 16, 20],  
 [5, 5, 10, 15, 20, 25]]  

# String Questions 

Question 1 — Reverse a String  
Problem Statement  
Write a program to reverse a given string.  
Input  
A string S.  
Output  
Print the reversed string.  
  
Examples 
Example 1:  
Input: hello  
Output: olleh  
Example 2:  
Input: I am iron man
Output: nam nori ma I  

In [2]:
s=input("Enter any words : ")
print(s[::-1])

nam nori ma i


Question 2 — Check if a String is Palindrome  
Problem Statement  
Given a string, check whether it is a palindrome. A string is a palindrome if it reads the same forward and backward.
Input  
A string S.  
Output  
Print Palindrome if S is a palindrome; otherwise print Not palindrome.
Examples  
Example 1:  
Input: ABCDCBA  
Output: Palindrome  
Example 2:  
Input: TAKE U FORWARD  
Output: Not Palindrome  

In [4]:
s=input("Enter any string : ")
a=s[::-1]
if a==s:
    print("Palindrome")
else:
    print("Not Palindrome")

Not Palindrome


Question 3 — Count Vowels, Consonants and Spaces  
Problem Statement  
Given a string, count the number of vowels, consonants, and spaces in it.  
Input  
A string S.  
Output  
Print the count of vowels, consonants, and spaces.  
Examples  
Example 1:  
Input: Take u forward is Awesome  
Output: Vowels: 10, Consonants: 11, White spaces: 4  
Example 2:  
Input: India won the cricket match  
Output: Vowels: 8, Consonants: 15, White spaces: 4  

In [14]:
s=input("Enter any sentences : ")
vowel="aeiouAEIOU"
v=0
c=0
sp=0

for i in s:
    if i in vowel:
        v+=1
    elif i ==" ":
        sp+=1
    else:
        c+=1
    # print(i)

print(f"Vowels : {v}, Consonants : {c}, White space : {sp}")


Vowels : 8, Consonants : 15, White space : 5


Question 4 — Find the ASCII Value of a Character  
  
Problem Statement:  
Given a character, find its ASCII value.
  
Input
A single character C.  
Output  
Print the ASCII value of C.  
  
Examples  
Example 1:  
Input: A  
Output: 65  
Example 2:  
Input: e  
Output: 101  

In [20]:
s=input("Enter any character : ")
print(ord(s))

101


Question 5 — Remove All Vowels from a String  
Problem Statement  
Given a string, remove all vowels from the string.  
Input  
A string S.  
Output  
Print the string after removing vowels.  
Examples  
Example 1:  
Input: take u forward
Output: tk frwrd   
Example 2:  
Input: I am very happy today  
Output: m vry hppy tdy  

In [2]:
s=input("Enter the string : ")
v="aeiouAEIOU"
out=""
for i in s:
    if i not in v:
        out=out+i

print(f"{out}")

 m vry hppy tdy 


Question 6 — Remove Spaces from a String  
Problem Statement  
Given a string, remove all spaces from the string.  
Input  
A string S.  
Output  
Print the string after removing spaces.  
Examples  
Example 1:  
Input: take U forward  
Output: takeUforward  
Example 2:  
Input: How are you doing  
Output: Howareyoudoing  

In [6]:
s=input("Enter any string: ")
out=" "
for i in s:
    if i!=" ": 
        out+=i
print(out)
    

 fbas


Question 7 — Remove Characters Except Alphabets  
Problem Statement  
Given a string, remove every character except alphabetic characters.  
Input  
A string S containing letters, digits, spaces, and/or special characters.  
Output  
Print only the alphabetic characters, preserving their original order.  
Examples  
Example 1:  
Input: take12% *&u; ^$#forward  
Output: takeuforward  
Example 2:  
Input: Java&C;++  
Output: JavaC  


In [12]:
st=input("Enter any sentences : ")
out=""
for i in st:
    if i.isalpha():
        out+=i
    
print(out)

javaC


Question 8 — Remove Brackets from an Algebraic  
Expression  
Problem Statement  
Given an algebraic expression, remove all bracket characters without changing the order of the remaining  
characters.  
Input  
An algebraic expression S.  
Output  
Print the expression after removing brackets.  
Examples  
Example 1:  
Input: a+((b-c)+d)  
Output: a+b-c+d  
Example 2:  
Input: (((a-b))+c)  
Output: a-b+c  

In [14]:
n=input("Enter any expression : ")
bra="()[]{}"
out=""
for i in n:
    if i not in bra:
        out+=i

print(out)


a-b+c


Question 9 — Sum of the Numbers in a String  
Problem Statement  
Given a string, calculate the sum of all numbers in it. Multiple consecutive digits are considered one number.    
Input  
A string S containing letters and digits.  
Output  
Print the sum of all numbers present in S.  
Examples  
Example 1:  
Input: 123xyz  
Output: 123  
Example 2:  
Input: 1xyz23  
Output: 24  

In [ ]:
# s=input("Enter any string : ")
# nu="0123456789"
# l=[]
# out=''
# for i in s:
#     if i in nu:
#         out+=i
#     elif i not in nu:
#         d=int(out)
#         continue
#     else:
#         out=""
#         if i in nu:
#             out+=i

# sum=d+int(out)
# print(sum)



    

    


124


Question 10 — Capitalize First and Last Character of Each Word  
Problem Statement  
Given a string, capitalize the first and last character of every word.  
Input  
A string S containing words separated by spaces.  
Output  
Print the modified string.  
Examples  
Example 1:  
Input: take u forward is awesome  
Output: TakE U ForwarD IS AwesomE  
Example 2:  
Input: I am a boy  
Output: I AM A BoY  

In [9]:
s=input("Enter any string : ")
out=""
p=s.split()
for i in p:
    out+=" "
    for j in range(0,len(i)):
        if j==0 or j==len(i)-1:
            out+=i[j].upper()
        else:
            out+=i[j]
print(out)
        

 I AM A BoY


Question 11 — Calculate Frequency of Characters  
Problem Statement  
Given a string, calculate the frequency of every character in the string.  
Input  
A string S.  
Output  
Print each character and its frequency.  
Examples  
Example 1:  
Input: takeuforward  
Output: a2 d1 e1 f1 k1 o1 r2 t1 u1 w1  
Example 2:  
Input: articles  
Output: a1 c1 e1 i1 l1 r1 s1 t1  


In [24]:
s=input("Enter any string : ")
p=sorted(s)
d={}
for i in p:
    d[i]=d.get(i,0)+1

a=list(d.keys())
b=list(d.values())
for i in range(0, len(a)):
    print(f"{a[i]}{b[i]}",end=" ")

a1 c1 e1 i1 l1 r1 s1 t1 

Question 12 — Find Non-Repeating Characters  
Problem Statement  
Given a string, print the characters that occur exactly once.  
Input  
A string S.  
Output  
Print all non-repeating characters in their original order.  
Examples  
Example 1:  
Input: google  
Output: l e  
Example 2:  
Input: yahoo  
Output: y a h  

In [4]:
s=input("Enter any string: ")
out=""
for i in s:
    if s.count(i)==1:
        out+=i
        print(i, end=" ")
# print(out)
   


y a h 

Question 13 — Check if Two Strings are Anagrams  
Problem Statement  
Given two strings, check whether they are anagrams of each other.  
Input  
Two strings S1 and S2.  
Output  
Print true if they are anagrams; otherwise print false.  
Examples  
Example 1:  
Input: CAT, ACT  
Output: true  
Example 2:  
Input: RULES, LESRT  
Output: false  


In [6]:
s1=input("Enter first string : ")
s2=input("Enter second string : ")
f=0
for i in s1:
    if i in s2:
        f+=1

if f==len(s1):
    print(True)
else:
    print(False)

False


Question 14 — Find the Maximum Occurring Character  
Problem Statement  
Given a string, return a character that occurs the maximum number of times. If multiple characters tie, return any one of them.  
Input  
A string S.  
Output  
Print one character having maximum frequency.  
Examples  
Example 1:    
Input: takeuforward  
Output: a  
Example 2:  
Input: apple  
Output: p  

In [11]:
s=input("Enter any string : ")
out=""
for i in s:
    if s.count(i)>=2:
        if i not in out:
            out=i
            break
print(out)



a


Question 15 — Remove All Duplicate Characters  
Problem Statement  
Given a string, remove repeated occurrences so that each character appears only once, preserving the first occurrence order.  
Input  
A string S.  
Output  
Print the resulting string.  
Examples  
Example 1: 
Input: bcabc  
Output: bca  
Example 2:  
Input: cbacdcbc  
Output: cbad  

In [13]:
s=input("Enter any string : ")
out=""
for i in s:
    if i not in out:
        out+=i
print(out)

cbad


Question 16 — Print All Duplicate Characters with Counts  
Problem Statement  
Given a string containing lowercase letters a to z, print characters that occur more than once along with their counts.  
Input  
A lowercase string S.  
Output  
Print every character whose frequency is greater than 1 and its count. If none exists, print an empty result.  
Examples  
Example 1:  
Input: sinstriiintng  
Output: i:4, n:3, s:2, t:2  
Example 2:  
Input: abcde  

In [17]:
s=input("Enter any string : ")
p=sorted(s)
d={}
for i in p:
    if p.count(i)>=2:
        d[i]=d.get(i,0)+1

a=list(d.keys())
b=list(d.values())

for i in range(0,len(a)):
    print(f"{a[i]}:{b[i]}", end=" ")

i:4 n:3 s:2 t:2 

Question 17 — Remove Characters from First String Present in Second.  
Problem Statement  
Given two strings, remove from the first string every character that is present in the second string.  
Input  
Two strings S1 and S2.  
Output  
Print the remaining characters of S1.  
Examples  
Example 1:  
Input: abcdef, cefz  
Output: abd  
Example 2:  
Input: xyzpw, lmno  
Output: xyzpw  

In [21]:
s1=input("Enter first string : ")
s2=input("Enter second string : ")

out=""
for i in s1:
    if i not in s2:
        out+=i

if len(out)==0:
    print(s1)
else:
    print(out)

abd


Question 18 — Change Every Letter to the Next Alphabet  
Problem Statement  
Replace every alphabetic character with the next letter in the alphabet. After z, continue with a; preserve letter case.  
Input  
A string S.  
Output  
Print the transformed string.  
Examples  
Example 1:  
Input: abcdxyz  
Output: bcdeyza  
Example 2:   
Input: Java  
Output: Kbwb  

In [ ]:
s=input("Enter any string : ")
out=""
for i in s:
    if i in "zZ":
        out+="a"
    else:
        d=ord(i)
        out=out+chr(d+1)
print(out)

Kbwb


Question 19 — Find the Largest Word in a String  
Problem Statement  
Given a string, find the longest word. If multiple words have the same maximum length, return the first one.  
Input  
A string containing words separated by spaces.  
Output  
Print the longest word.  
Examples  
Example 1:  
Input: Google Doc  
Output: Google  
Example 2:  
Input: Microsoft Teams  
Output: Microsoft  


In [42]:
s=input("Enter any string : ")
p=s.split()
x=[]
for i in p:
    x.append(len(i))

v=x.index(max(x))
print(p[v])

Microsoft


Question 20 — Sort Characters in a String  
Problem Statement  
Sort the alphabetic characters of a string in ascending order.  
Input  
A string containing alphabetic characters.  
Output  
Print the sorted string.  
Examples  
Example 1:  
Input: zxcbg  
Output: bcgxz  
Example 2:  
Input: edcba  
Output: abcde  


In [46]:
s=input("Enter any string : ")
p=sorted(s)
l="".join(p)
print(l)

abcde


Question 21 — Count the Number of Words  
Problem Statement  
Given a string, count the number of words. A word is a sequence of non-space characters.  
Input  
A string S.  
Output  
Print the number of words.  
Examples  
Example 1:   
Input: Hello  
Output: 1  
Example 2:  
Input: Hi there  
Output: 2  

In [48]:
s=input("Enter any string : ")
p=s.split()
print(len(p))

2


Question 22 — Find the Word with Highest Repeated-Letter Frequency  
Problem Statement  
Find the word containing a letter with the highest frequency among all words. If no word contains a  
repeated letter, print -1.  
Input  
A string containing words separated by spaces.  
Output  
Print the first word with the highest repeated-letter frequency, or -1.  
Examples  
Example 1:  
Input: abcdefghij google microsoft  
Output: google  
Example 2:  
Input: cameron blue  
Output: -1  


In [ ]:
# s=input("Enter any string : ")
# p=s.split()

# for i in p:
#     a=0
#     count=0
#     for j in i:
#         if i.count(j)>=2:
#             count+=1


#     if count>=2:
#         a+=1
#         rep=i
    

# if rep in p and a>=2:
#     print(rep)
# else:
#     print(-1)




-1


Question 23 — Change Case of Each Character  
Problem Statement  
Change lowercase letters to uppercase and uppercase letters to lowercase. Leave 
on-alphabetic characters unchanged.  
Input  
A string S.  
Output  
Print the string after changing the case of every alphabetic character.  
Examples  
Example 1:  
Input: javA  
Output: JAVa  
Example 2:  
Input: take u forward IS Awesome  
Output: TAKE U FORWARD is aWESOME  


In [64]:
s=input("Enter any string : ")
out=""
for i in s:
    if i.islower():
        out+=i.upper()
    else:
        out+=i.lower()
print(out)
        

JAVa


Question 24 — Concatenate Two Strings  
Problem Statement  
Concatenate two strings and form one combined string while preserving their original order.  
Input  
Two strings S1 and S2.  
Output  
Print the concatenated string.  
Examples  
Example 1:  
Input: hello, world  
Output: helloworld  

In [ ]:
s1=input("Enter first string : ")
s2=input("Enter second string : ")
print(f"{s1+s2}")

helloworld


Question 25 — Find the Position of a Substring  
Problem Statement  
Given a text and a pattern, find the first occurrence of the pattern in the text.  
Input  
Two strings: text and pattern.  
Output  
Print the starting index of the first occurrence, or -1 if the pattern is not present.  
Examples  
Example 1:  
Input: takeuforward, forward  
Output: 5  
Example 2: 
Input: hello, az  
Output: -1  


In [75]:
text=input("Enter any string : ")
pat=input("Enter pattern : ")

if pat in text:
    print(text.index(pat))
else:
    print(-1)
    


5


Question 26 — Reverse Words in a String  
Problem Statement  
Reverse the order of words in a string. The output must contain exactly one space between words and no  
leading or trailing spaces.  
Input  
A string S containing letters, digits, and spaces.  
Output  
Print the words in reverse order with single spaces.  
Examples  
Example 1:  
Input: welcome to the jungle  
Output: jungle the to welcome  
Example 2:  
Input: amazing coding skills  
Output: skills coding amazing  


In [78]:
s=input("Enter any string : ")
p=s.split()[::-1]
d=" ".join(p)
print(d)

skill coding amazing


Question 27 — Count Digits, Letters and Special Characters  
Problem Statement  
Given a string, count how many characters are digits, alphabetic characters, and special characters.  
Input  
A string S.  
Output  
Print the three counts.  
Examples  
Example 1:  
Input: Hello123!  
Output: Letters: 5, Digits: 3, Special: 1  
Example 2:  
Input: A@1#  
Output: Letters: 1, Digits: 1, Special: 2  

In [84]:
s=input("Enter any string : ")
l=0
d=0
sp=0
for i in s:
    if i.isalpha():
        l+=1
    elif i.isdigit():
        d+=1
    else:
        sp+=1
print(f"Letter: {l}, Digit: {d}, Special: {sp} ")

Letter: 1, Digit: 1, Special: 2 


Question 28 — Count Uppercase and Lowercase Characters  
Problem Statement  
Given a string, count the number of uppercase and lowercase alphabetic characters.  
Input  
A string S.  
Output  
Print the uppercase and lowercase counts.  
Examples  
Example 1:  
Input: Hello World  
Output: Uppercase: 2, Lowercase: 8  
Example 2:  
Input: ABCdef  
Output: Uppercase: 3, Lowercase: 3  


In [87]:
s=input("Enter any string : ")
u=0
l=0
for i in s:
    if i.isupper():
        u+=1
    elif i.islower():
        l+=1

print(f"Uppercase: {u}, Lowercase: {l}")

Uppercase: 4, Lowercase: 2


Question 29 — Remove Duplicate Words from a String  
Problem Statement  
Given a sentence, remove repeated words while preserving the first occurrence of each word.  
Input  
A sentence S.  
Output  
Print the sentence containing each word only once.  
Examples  
Example 1:  
Input: this is is a test test  
Output: this is a test  
Example 2:  
Input: hello hello world  
Output: hello world  


In [89]:
s=input("Enter the string : ")
p=s.split()
l=[]
for i in p:
    if i not in l:
        l.append(i)
d=" ".join(l)
print(d)

hello world


Question 30 — Count Occurrences of a Given Character  
Problem Statement  
Given a string and a character, count how many times the character occurs in the string.  
Input  
A string S and a character C.  
Output  
Print the number of occurrences of C.  
Examples  
Example 1:  
Input: banana, a  
Output: 3  
Example 2:  
Input: hello, l  
Output: 2  

In [94]:
s=input("Enter any string : ")
a=input("Enter any character : ")
print(s.count(a))

3


Question 31 — Replace Every Space with a Hyphen  
Problem Statement  
Given a string, replace every space character with a hyphen.  
Input  
A string S.  
Output  
Print the modified string.  
Examples  
Example 1:  
Input: hello world  
Output: hello-world  
Example 2:  
Input: Python is easy  
Output: Python-is-easy  

In [97]:
s=input("Enter any string : ")
p=s.split()
d="-".join(p)
print(d)


hello-world


Question 32 — Check if a String Contains Only Digits  
Problem Statement  
Given a string, determine whether every character in the string is a digit.  
Input  
A string S.  
Output  
Print True if all characters are digits; otherwise print False.  
Examples  
Example 1:  
Input: 12345  
Output: True  
Example 2:  
Input: 123a45  
Output: False  


In [100]:
s=input("Enter any string : ")
if s.isdigit():
    print(True)
else:
    print(False)



True


Question 33 — Check if a String Contains Only Alphabets  
Problem Statement  
Given a string, determine whether every character is an alphabetic character.  
Input  
A string S.  
Output  
Print True if all characters are alphabets; otherwise print False.  
Examples  
Example 1:  
Input: Hello  
Output: True  
Example 2:  
Input: Hello123  
Output: False  

In [102]:
s=input("Enter any string : ")
if s.isalpha():
    print(True)
else:
    print(False)

False


Question 34 — Find the First Non-Repeating Character  
Problem Statement  
Given a string, find the first character whose frequency is exactly one.  
Input  
A string S.  
Output  
Print the first non-repeating character, or -1 if none exists.  
Examples  
Example 1:  
Input: swiss  
Output: w  
Example 2:  
Input: aabbcc  
Output: -1  

In [105]:
s=input("Enter any string : ")
for i in s:
    if s.count(i)==1:
        a=i
        break

if a in s:
    print(a)
else:
    print(-1)
    

m


Question 35 — Find the First Repeating Character  
Problem Statement  
Given a string, find the first character that appears more than once.  
Input  
A string S.  
Output  
Print the first repeating character, or -1 if no character repeats.  
Examples  
Example 1:  
Input: swiss  
Output: s  
Example 2:  
Input: abcdef  
Output: -1  

In [108]:
s=input("Enter any string : ")
for i in s:
    if s.count(i)>=2:
        a=i
        break
if a in s:
    print(a)
else:
    print(-1)

a


Question 36 — Remove a Given Character from a String  
Problem Statement  
Given a string and a character, remove all occurrences of that character.  
Input  
A string S and a character C.  
Output  
Print the string after removing C.  
Examples  
Example 1:  
Input: banana, a  
Output: bnn  
Example 2:  
Input: hello, l  
Output: heo  

In [112]:
s=input("Enter any string : ")
c=input("Enter any character : ")
out=""
for i in s:
    if i!=c:
        out+=i
print(out)

amandp


Question 37 — Count Occurrences of a Substring  
Problem Statement  
Given a string and a substring, count how many times the substring occurs without overlapping
occurrences.  
Input  
Two strings: text and pattern.  
Output  
Print the number of occurrences of pattern in text.  
Examples  
Example 1:  
Input: aaaa, aa  
Output: 2 
Example 2:  
Input: hello hello, hello  
Output: 2  


In [116]:
t=input("Enter any string : ")
p=input("Enter pattern : ")
print(t.count(p))

2


Question 38 — Check if One String is a Rotation of Another  
Problem Statement  
Given two strings, check whether the second string can be obtained by rotating the first string.  
Input  
Two strings S1 and S2. 
Output  
Print True if S2 is a rotation of S1; otherwise print False.  
Examples  
Example 1:  
Input: ABCD, CDAB  
Output: True  
Example 2:  
Input: ABCD, ACBD  
Output: False  

In [118]:
# s1=input("Enter first string : ")
# s2=input("ENTER second string :")
# for j in range(0,len(s1)):
#     i=0
#     s1[i],s1[i+1]=s1[i+2],s1[i+3]
# print(s1)
   
       

Question 39 — Find the Shortest Word in a String  
Problem Statement  
Given a string containing words, find the shortest word. If multiple words have the same minimum length, return the first one.  
Input  
A string containing words separated by spaces.  
Output  
Print the shortest word.  
Examples  
Example 1:  
Input: I love Python  
Output: I  
Example 2:  
Input: code in Java  
Output: in  

In [128]:
s=input("Enter any string : ")
p=s.split()
c=[]
for i in p:
    c.append(len(i))

m=c.index(min(c))
print(p[m])


i


Question 40 — Check if a String is a Valid Identifier  
Problem Statement  
Given a string, determine whether it can be used as a valid Python identifier.  
Input  
A string S.  
Output  
Print Valid if S is a valid Python identifier; otherwise print Invalid.  
Examples  
Example 1:  
Input: student_name  
Output: Valid  
Example 2:  
Input: 123student  
Output: Invalid  

In [130]:
s=input("Enter any string : ")
if s.isidentifier():
    print("valid")
else:
    print("Invalid")

Invalid


# Extra question 

In [ ]:
n1=int(input("Enter first number : "))
n2=int(input("Enter second number : "))
small=0
if n1>n2:
    small=n2
else:
    small=n1

for i in range(1,small+1):
    if n1%i==0 and n2%i==0:
        gcd=i

print(f"The greatest common divisor of {n1} and {n2} is {gcd}.")

In [38]:
# WAP TO GET THE FOLLOWING OUTPUT 
# INPUT : "push maadi kushi padi"
# OUTPUT: {"push":"ph","maddi":"a","kushi":"s","padi":"pi"}


s=input("Enter any string : ")
p=s.split()
d={}
for i in p:
    if len(i)%2==0:
        o=i[0]+i[-1]
        d[i]=d.get(i,"")+o
    else:
        mid=len(i)//2
        d[i]=d.get(i,"")+i[mid]

print(d)
    
    


{'push': 'ph', 'maddi': 'd', 'kushi': 's', 'padi': 'pi'}


In [43]:
# WAP TO GET THE FOLLOWING OUTPUT:
# S="HAI HELLO"
# OUT={"HAI":"AI","HELLO":"EO"}

s=input("Enter any string : ")
v="aeiouAEIOU"
d={}
p=s.split()
for i in p:
    o=""
    for j in range(0,len(i)):
        if i[j] in v:
            o=o+i[j]
    d[i]=d.get(i,"")+o

print(d)


{'Hello': 'eo', 'Amandeep': 'Aaee'}


In [50]:
# WAP TO GET THE FOLLOWING OUTPUT 
# S=["jiocinema.com","file.py","web.html","amazon.com","www.org"]
# out=["com","py","html","org"]

s=eval(input("Enter any string : "))
out=[]
for i in s:
    ex=i.split(".")[-1]
    if ex not in out:
        out.append(ex)
print(out)

['com', 'py', 'html', 'org']


In [ ]:
# wap to get the following output
# s=["jiocinema.com","file.py","web.html","amazon.com","www.org"]
# out=["com":["jiocinema","amazon"],"py":["file","python"],"html":["web"],"org":["www"]]

s = ["jiocinema.com", "file.py", "python.py", "web.html", "amazon.com", "www.org"]

out = {}

for i in s:
    name, ex= i.rsplit(".", 1)

    if ex not in out:
        out[ex] = []

    out[ex].append(name)

print(out)



# wap to get the following output

{'com': ['jiocinema', 'amazon'], 'py': ['file', 'python'], 'html': ['web'], 'org': ['www']}


: 